components/camera/videosource/webcam_test.go

```diff
@@ -54,3 +59,51 @@ func TestDiscoveryWebcam(t *testing.T) {
 	test.That(t, respProps[0].FrameFormat, test.ShouldResemble, "some format")
 	test.That(t, respProps[0].FrameRate, test.ShouldResemble, float32(30))
 }
+
+// Testing FrameRate camera property
+
+type mockVideoSource struct {
+	frameRate float32
+}
```

Comment:
Can we use `testutils/inject`'s injected cameras to mock up the camera instead? It's more in line wiht our in-rdk testing style.

Reply:
> if we use inject.Camera will this still be considered a mock "webcam"? Just trying to clarify my understanding here

Yes, the injects will need you to fulfill their interfaces otherwise they will be nil, they're just easier to code because they give you helpers for adding a resource name and required interface functions or call the underlying function if there is a real implementation being wrapped. 

> If we use inject.Camera to create a "mockCam" and then set that mockCam as the exposedProjector inside the fakeMonitoredWebcam, then when we test the properties of the exposedProjector we are just testing the properties of the inject.Camera 

we'd have to have the injected/mocked camera actually use an image with projected values and verify that they are what we expect. `image_file` does this with it's static source. The parts of webcam that are hard to test are its creator function since it's relying on a video reader connected to a serial port. This, in essence, is I think what the virtual camera tries to do but I have no clue if those tests work because they don't run. You're right, mocks don't test this capability, and in general for hardware-in-the-loop functionality I like canaries more. However, for code that is essentially math (generating a pointcloud from a 2d image) or informative (frame rate from GetProperties), we can have simple data fed into our mock and test that capability.